library/src/main/java/com/datatorrent/lib/db/jdbc/AbstractJdbcPollInputOperator.java

```diff
@@ -282,23 +330,44 @@ public void deactivate()
     store.disconnect();
   }
 
+  protected Object extractKey(ResultSet rs) throws SQLException
+  {
+    return rs.getObject(this.key);
+  }
+
   /**
    * Execute the query and transfer results to the emit queue.
    * @param preparedStatement PreparedStatement to execute the query and fetch results.
    */
-  protected void insertDbDataInQueue(PreparedStatement preparedStatement) throws SQLException, InterruptedException
+  protected int insertDbDataInQueue(PreparedStatement preparedStatement) throws SQLException, InterruptedException
   {
+    int resultCount = 0;
     preparedStatement.setFetchSize(getFetchSize());
     ResultSet result = preparedStatement.executeQuery();
     if (result.next()) {
       do {
-        while (execute && !emitQueue.offer(getTuple(result))) {
+        T obj = getTuple(result);
```

Comment:
Can we check the result for null and skip it ? Then we won't need to null check inside the loop in emitTuples().